pipeline/relation_embedder/path_concatenator/README.md

```diff
@@ -0,0 +1,54 @@
+# Path Concatenator
+
+## What?
+
+Unlike records from other systems each store their full path to the root of their hierarchy,
+records from Sierra only know their immediately adjacent parent/child relations.
+
+This makes it impossible to construct a full hierarchical path just from the data in a single record (unless that
+record is either the root or a child of the root).
+
+When a Sierra record in a hierarchy is discovered, it uses the path_concatenator to complete its place in the hierarchy
+if necessary.
+
+See [RFC 046: Transitive hierarchies in Sierra](https://github.com/wellcomecollection/docs/tree/main/rfcs/046-transitive-sierra-hierarchies)
+for more detail.
+
+## How?
+
+1. Take the first and last segments of the path
+    * e.g. given a path, `root/branch/leaf` - it will use `root`, and `leaf`.
+2. Run a wildcard search for records whose last segment matches the first segment of this record.
+    * e.g. `*/root`
+    * This should only match, at most, one record, if there are more, log an error and do nothing.
+    * If it does not match any records, continue.
+3. Replace the first segment in this record with the collectionPath of that record.
+    * e.g. this record is `d/e/f`, there exists `b/c/d`, the collectionPath for this record becomes `b/c/d/e/f`
+4. Run a term search for records with a collectionPath matching the last segment
+    * collectionPath is a path_hierarchy, so in the example above, this will match any records with a path that start with `leaf`
+5. Replace the first segment in those collectionPath, with this record's collectionPath
+    * e.g. a path `leaf/1/2` would become `root/branch/leaf/1/2`
+6. Notify downstream (batcher) of all changed paths.
+7. Notify downstream (batcher) of the current path if it is unchanged.
+
+This is a deliberately ordered list of steps, so that the current record *first* gets its own path embellished, *then*
+goes on to embellish the paths of its children with that new path.
+
+When writing to the database, pay attention to version.  If it is not what is expected,
```

Comment:
What do you mean by `version` here? It's a bit of an overloaded term and I'm not convinced it's as important as we think.

Reply:
I mean version as in here: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#docs-index-api-query-params

I agree that it's unlikely to be a problem.